.github/workflows/specs.yml

```diff
@@ -3,15 +3,41 @@ on: [push, pull_request]
 permissions:
   contents: read # to fetch code (actions/checkout)
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
I don't think we should cancel CI jobs

Reply:
Let me explain it. Canceling is only when we push new changes to the same branch. This allows you to save running time on GitHub if you make frequent changes. This does not affect CI if work is carried out on different branches.

Reply:
I know what it does, but for instance it could cancel a CI run on master if 2 PRs are merged quickly one after another, and that seems bad (could make it harder to find a faulty change) + there is not really enough CI load on this repo to need to cancel anything.

Reply:
We can exclude cancel for branch master. WDYT?

Reply:
But then that prevents to run CI jobs on different commits on master in parallel, which is unnecessary. Please remove this section entirely, there is no need to change this.

Reply:
I feel like removing will hurt the project, but I'll adapt.

Reply:
Because CI would start with a delay? I've never observed it on rvm/rvm, so I think it's a not an issue in practice (the default numbers of runners is pretty high).
And for the scenario I mentioned for master it's actually worse, adding a constraint which is not needed.
Another downside is e.g. when working on CI one might push more changes but still be interested in older in-progress CI jobs (e.g. I still want to see macOS CI results when touching common logic). This would cancel them and make it very annoying for those cases, which IMO outweighs any possible gain from this.

Reply:
The way I see it is that if I make changes, I would like to minimize the time in which I receive information that the tests have passed and I don't want to wait until the entire CI queue is completed. I completely agree that the master branch should be unique in this case. But maybe it won't be such a big problem. If it hurts, we can consider the proposed changes.